### PR TITLE
[localstorage] Small refactor

### PR DESCRIPTION
**Pull request by marechal-p**
Refactors the LocalStorage implementations to use inversify.

Signed-off-by: Paul Maréchal <paul.marechal@ericsson.com>

### CODE REVIEW

File: packages/workspace/src/browser/workspace-frontend-module.ts [hunk 1]
```diff
@@ -54,7 +54,9 @@ export default new ContainerModule((bind: interfaces.Bind, unbind: interfaces.Un
     bind(CommandContribution).to(WorkspaceCommandContribution).inSingletonScope();
     bind(MenuContribution).to(FileMenuContribution).inSingletonScope();
 
-    rebind(StorageService).to(WorkspaceStorageService).inSingletonScope();
+    bind(WorkspaceStorageService).toSelf().inSingletonScope();
+    rebind(StorageService).toService(WorkspaceStorageService);
+    console.log('WS SERVICE', WorkspaceService);
```
Comment (akosyakov) [inline packages/workspace/src/browser/workspace-frontend-module.ts]: should be removed?

File: packages/workspace/src/browser/workspace-storage-service.ts [hunk 2]
```diff
@@ -28,18 +28,23 @@ export class WorkspaceStorageService implements StorageService {
...
     private prefix: string;
     private initialized: Promise<void>;
-    protected storageService: StorageService;
 
-    constructor( @inject(WorkspaceService) protected workspaceService: WorkspaceService,
-        @inject(ILogger) protected logger: ILogger) {
+    @inject(LocalStorageService) protected storageService: StorageService;
+    @inject(WorkspaceService) protected workspaceService: WorkspaceService;
+
+    constructor(
```
Comment (akosyakov) [inline packages/workspace/src/browser/workspace-storage-service.ts]: Could be removed?

**Comment by akosyakov** [comment]
Could you elaborate on why it is needed? I think it was done like that on purpose that no one uses `LocalStorageService` directly. Clients should use `StorageService`.

**Comment by marechal-p** [comment]
It is just an issue that someone in our team had trouble with, injection requests are defined (`@inject(...)`), but because the class was used outside inversify the instanciation was not done properly.

So if we use a component that has dependencies, should we make the owner of this component use the same dependencies to then inject it by hand into the instance made without inversify ?

**Pull request merged**